packages/cli/src/commands/api-generator/watchMode/watchMode.ts

```diff
@@ -0,0 +1,60 @@
+import console from "node:console";
+
+import { type ChildProcessWithoutNullStreams, spawn } from "child_process";
+import { watch } from "chokidar";
+
+import { handleChildProcess } from "./handleChildProcess";
+
+/**
+ * Watch mode for the generator.
+ *
+ * Watches the `src` directory for changes and triggers child processes of generator for the changed file, to process it.
+ */
+export const watchMode = async () => {
+    /**
+     * Collection of last ChildProcesses for each file
+     */
+    const childProcesses: Record<string, ChildProcessWithoutNullStreams> = {};
+    watch("./src", {
+        awaitWriteFinish: {
+            stabilityThreshold: 300,
+            pollInterval: 200,
+        },
+        ignored: (path, stats) => {
+            if (stats?.isFile()) {
+                return !path.endsWith("entity.ts");
+            }
+            return false;
+        },
+    })
+        .on("change", async (path) => {
```

Comment:
How does chokidar handle consecutive writes to the same file? Would it be possible to wait until the child process has finished? The we wouldn't need to kill "old" child processes.

Reply:
what problem do you see with killing?

Reply:
Probably that it may be as unreliable as the "old" way we restarted the Vite dev server, resulting in tangling Node processes. And doing lot's of useless work 🤷🏼‍♂️ 

But if we can reliably kill child process than there's no real problem.

Reply:
killing processes works perfectly fine, when done correctly.

What can cause problems are multiple child processes (a tree), that can happen for example with `npm run xxx` that itself starts another child process. If you kill the `npm run` process, it's child process doesn't receive a kill signal. (This is different from Ctrl+C on the cli, which gets sent to all child processes). (I had those issues in dev-process-manager)

When you have a single child process (should be the case here, though it should be tested explicitly) this won't be an issue.